internal/provider/datasources/data_source_notification_channels.go

```diff
@@ -0,0 +1,183 @@
+package datasources
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+
+	"github.com/samber/lo"
+
+	"github.com/astronomer/terraform-provider-astro/internal/clients"
+	"github.com/astronomer/terraform-provider-astro/internal/clients/platform"
+	"github.com/astronomer/terraform-provider-astro/internal/provider/models"
+	"github.com/astronomer/terraform-provider-astro/internal/provider/schemas"
+	"github.com/astronomer/terraform-provider-astro/internal/utils"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+// Ensure provider defined types fully satisfy framework interfaces.
+var _ datasource.DataSource = &notificationChannelsDataSource{}
+var _ datasource.DataSourceWithConfigure = &notificationChannelsDataSource{}
+
+func NewNotificationChannelsDataSource() datasource.DataSource {
+	return &notificationChannelsDataSource{}
+}
+
+// notificationChannelDataSource defines the data source implementation.
+type notificationChannelsDataSource struct {
+	PlatformClient platform.ClientWithResponsesInterface
+	OrganizationId string
+}
+
+func (d *notificationChannelsDataSource) Metadata(
+	ctx context.Context,
+	req datasource.MetadataRequest,
+	resp *datasource.MetadataResponse,
+) {
+	resp.TypeName = req.ProviderTypeName + "_notification_channels"
+}
+
+func (d *notificationChannelsDataSource) Schema(
+	ctx context.Context,
+	req datasource.SchemaRequest,
+	resp *datasource.SchemaResponse,
+) {
+	resp.Schema = schema.Schema{
+		// This description is used by the documentation generator and the language server.
+		MarkdownDescription: "Notification Channels data source",
+		Attributes:          schemas.NotificationChannelsDataSourceSchemaAttributes(),
+	}
+}
+
+func (d *notificationChannelsDataSource) Configure(
+	ctx context.Context,
+	req datasource.ConfigureRequest,
+	resp *datasource.ConfigureResponse,
+) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	apiClients, ok := req.ProviderData.(models.ApiClientsModel)
+	if !ok {
+		utils.DataSourceApiClientConfigureError(ctx, req, resp)
+		return
+	}
+
+	d.PlatformClient = apiClients.PlatformClient
+	d.OrganizationId = apiClients.OrganizationId
+}
+
+func (d *notificationChannelsDataSource) Read(
+	ctx context.Context,
+	req datasource.ReadRequest,
+	resp *datasource.ReadResponse,
+) {
+	var data models.NotificationChannels
+
+	// Read Terraform configuration data into the model
+	resp.Diagnostics.Append(req.Config.Get(ctx, &data)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	params := &platform.ListNotificationChannelsParams{
+		Limit: lo.ToPtr(1000),
```

Comment:
is listing 1000 safe?

Reply:
It might be if there aren't too many joins

Reply:
yes 1000 is safe, we are just querying one table with no joins